lib/rubocop/cop/rails/foreign_key_name.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces that foreign keys are created with explicitly specified names.
+      #
+      # @example
+      #   # bad
+      #   add_foreign_key :articles, :authors
+      #   t.foreign_key :authors
+      #
+      #   # good
+      #   add_foreign_key :articles, :authors, name: :articles_authors_fk
+      #   t.foreign_key :authors, name: :articles_authors_fk
+      #
+      class ForeignKeyName < Base
+        extend AutoCorrector
+
+        MSG = 'Name your foreign keys explicitly instead of relying on Rails auto-generated FK names.'
+
+        TABLE_METHODS = %i[create_table change_table].freeze
+
+        def_node_matcher :foreign_key_name_candidate?, <<~PATTERN
+          {
+            (send nil? :add_foreign_key _ $_ $!(hash <(pair (sym :name) _) ...>) ?)
+            (send !nil? :foreign_key $_ $!(hash <(pair (sym :name) _) ...>) ?)
+          }
+        PATTERN
+
+        def on_send(node)
+          foreign_key_name_candidate?(node) do |to_table_node, options_node|
+            options_node = options_node.first
+
+            add_offense(node) do |corrector|
+              foreign_key_name = foreign_key_name(node, to_table_node, options_node)
+
+              if options_node
+                corrector.insert_after(options_node, ", name: :#{foreign_key_name}")
+              else
+                corrector.insert_after(node, ", name: :#{foreign_key_name}")
+              end
+            end
+          end
+        end
+
+        private
+
+        def foreign_key_name(node, to_table_node, options_node)
+          column = "#{to_table_node.value.to_s.singularize}_id"
+
+          if options_node
+            column_pair = options_node.pairs.find { |p| p.key.value.to_sym == :column }
+            column = column_pair.value.value if column_pair
+          end
+
+          from_table = from_table(node)
+          "#{from_table}_#{column}_fk"
```

Comment:
What if we return a Symbol here, and then just `insert_after(node, ", name: foreign_key_name)` above with no string interpolation and `:"` conversion?

Reply:
Tried your suggestion:
```ruby
insert_after(node, name: foreign_key_name)
```

But got this:
```ruby
+add_foreign_key :articles, :authors{:name=>:articles_author_id_fk}
```

It just appended "inspected" value. Seems like we need to explicitly build an appended source string.